committer/src/main.rs

```diff
@@ -34,6 +35,11 @@ async fn main() -> Result<()> {
 
     let metrics_registry = Registry::default();
 
+    let (ethereum_rpc, eth_health_check) =
+        create_l1_adapter(&config, &internal_config, &metrics_registry).await?;
+
+    config.set_commit_interval(ethereum_rpc.commit_interval()?);
```

Comment:
Oh, I see your dilemma. You'd like for `config` to carry the commit interval deeper into the system. 

I dislike it being public. If you can find a way to not have it be configurable by the user then cool. 

Maybe something like serde ignore a field? Make sure you test it out though. Don't want a stray env variable to set the field even if it going to be overwritten later on.

Or just carry this along separately without it being part of the config.

Reply:
it wont be part of config